flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -4303,4 +4310,40 @@ private void ackCheckpoint(
                                                 .build()))),
                 "test");
     }
+
+    static class OperatorSubtaskStateMock {
+        OperatorSubtaskState subtaskState;
+        OperatorStateHandle managedOpHandle;
+        OperatorStateHandle rawOpHandle;
+
+        OperatorSubtaskStateMock() throws Exception {
+            this.managedOpHandle = mock(OperatorStreamStateHandle.class);
+            this.rawOpHandle = mock(OperatorStreamStateHandle.class);
+            OperatorSubtaskState subtaskState =
+                    OperatorSubtaskState.builder()
+                            .setManagedOperatorState(managedOpHandle)
+                            .setRawOperatorState(rawOpHandle)
+                            .build();
+            this.subtaskState = spy(subtaskState);
+        }
+
+        public OperatorSubtaskState getSubtaskState() {
+            return this.subtaskState;
+        }
+
+        public void reset() {
+            Mockito.reset(managedOpHandle);
+            Mockito.reset(rawOpHandle);
+        }
+
+        public void verifyDiscard() throws Exception {
+            verify(managedOpHandle, times(1)).discardState();
+            verify(rawOpHandle, times(1)).discardState();
+        }
+
+        public void verifyNotDiscard() throws Exception {
+            verify(managedOpHandle, never()).discardState();
+            verify(rawOpHandle, never()).discardState();
+        }
+    }
```

Comment:
Can you rewrite this class without using `Mockito` (as per our [code style](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#avoid-mockito---use-reusable-test-implementations))? I would guess you can replace

```
this.managedOpHandle = mock(OperatorStreamStateHandle.class);
this.rawOpHandle = mock(OperatorStreamStateHandle.class);
```
with proper mocks implementing `org.apache.flink.runtime.state.OperatorStateHandle` interface that internally are checking if `discardState()` has been called or not (boolean flag) and exposing via `boolean isDiscarded()` method or something similar.

Reply:
updated. great instructions on the mock usage (or rather no-mock usage)!